include/rear-panel/core/messages.hpp

```diff
@@ -1,31 +1,132 @@
 #pragma once
+#include <algorithm>
 #include <cstdint>
 #include <variant>
 
-namespace messages {
-
-struct IncomingMessageFromHost {
-    const uint8_t* buffer;
-    const uint8_t* limit;
-};
+#include "common/core/bit_utils.hpp"
+#include "common/core/version.h"
+#include "rear-panel/core/bin_msg_ids.hpp"
+#include "rear-panel/core/binary_parse.hpp"
 
-// TODO(ryan): as part of RET-1307 create an auto gen of the ids like the can
-// messages and include them from another file
-enum class MessageType : uint16_t {
-    ECHO = 0x00,
-};
+// this is the max size of the double_buffer used in freertos_message_queue
+constexpr size_t MAX_MESSAGE_SIZE = 512U * 4;
```

Comment:
this is... pretty big. I think 1024 might be better since I think that's the max payload size for highspeed data payloads per https://www.beyondlogic.org/usbnutshell/usb3.shtml